js/boaexchange.js

```diff
@@ -0,0 +1,1021 @@
+        if ('address' in transaction) {
+            address = transaction['address']['address'];
+        } else {
+            address = transaction['address'];
+        }
+        let code = undefined;
+        let currencyId = this.safeString (transaction['coin'], 'code');
+        currency = this.safeValue (this.currencies_by_id, currencyId);
+        if (currency !== undefined) {
+            code = currency['code'];
+        } else {
+            code = this.commonCurrencyCode (currencyId);
+        }
+        return {
+            'info': transaction,
+            'id': transaction['txid'],
+            'txid': transaction['txid'],
+            'timestamp': transaction['created'],
+            'datetime': this.parse8601 (transaction['created']),
+            'addressFrom': undefined,
+            'address': address,
+            'addressTo': undefined,
+            'tagFrom': undefined,
+            'tag': undefined,
+            'tagTo': undefined,
+            'type': transaction,
+            'amount': transaction['amount'],
+            'currency': code,
+            'status': 'ok',
+            'updated': undefined,
+            'message': undefined,
+            'fee': undefined,
+        };
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.implodeParams (this.urls['api'], {
+            'hostname': this.hostname,
+        });
+        url += '/' + this.version + '/' + this.implodeParams (path, params);
+        params['limit'] = 500;
+        url += '?' + this.urlencode (params);
+        if (api === 'private') {
+            this.checkRequiredCredentials ();
+            let nonce = this.nonce ().toString ();
+            let signature = this.hmac (this.encode (nonce), this.encode (this.secret), 'sha256');
+            headers = {
+                'X-BOA-ENCRYPTED': signature,
+                'X-BOA-KEY': this.apiKey,
+                'X-BOA-NONCE': nonce,
+            };
+        }
+        return { 'url': url, 'method': method, 'body': body, 'headers': headers };
+    }
+
+    handleErrors (code, reason, url, method, headers, body, response) {
+        if (body[0] === '{') {
+            let data = this.safeValue (response, 'data');
+            let errors = this.safeValue (response, 'errors');
+            const feedback = this.id + ' ' + this.json (response);
+            if (errors !== undefined) {
+                const message = errors[0];
+                if (message in this.exceptions)
+                    throw new this.exceptions[message] (feedback);
+                throw new ExchangeError (url + method + ' an error occoured: ' + this.json (errors));
+            }
+            if (data === undefined)
+                throw new ExchangeError (this.id + ': malformed response: ' + this.json (response));
+        }
+    }
+
+    async request (path, api = '', method = 'GET', params = {}, headers = undefined, body = undefined) {
```

Comment:
And this method also doesn't add anything to the one that is in the base class, so we practically very rarely need to redefine it, and if you see it somewhere, it's mostly there for historical reasons. So, in this particular case, it can be safely removed as well.

Reply:
done